src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -42,6 +42,12 @@
 #include "src/core/lib/transport/static_metadata.h"
 #include "src/core/lib/transport/timeout_encoding.h"
 
+/* don't consider adding anything bigger than this to the hpack table */
+#define MAX_DECODER_SPACE_USAGE 512
+#define DATA_FRAME_HEADER_SIZE 9
+
+namespace { /* (Maybe-cuckoo) hpack encoder hash table implementation. */
```

Comment:
Does this comment apply to everything inside this anonymous namespace? What does "Maybe-cuckoo" mean?

Reply:
Yes, everything in that namespace is the refactored hash table.

Maybe-cuckoo - so right now, our hash table is sort of a cuckoo hash - I say cuckoo in that it implements one part of cuckoo hashing (secondary hashing buckets) but not the other part (rearranging elements to secondary buckets in a loop till things fit to handle collisions). 

I'm still seeing, from a perf. POV, if cuckoo hashing is better than having the fallback bucket be the next bucket linearly (which theoretically has better cache performance) vs. just not having fallback spots (and evicting in the case of collision). To make it easy to test, I have added a #define  to control whether we do our (sort-of) cuckoo hash or if we just evict immediately. Hence the name, maybe-cuckoo.